tests/v2/test_v2_session_lifecycle.py

```diff
@@ -65,7 +76,13 @@ async def handle(websocket: ServerConnection) -> None:
                     state["ipv4_laddr"] = pair
         serve_forever = asyncio.create_task(server.serve_forever())
         yield None
-        serve_forever.cancel()
+        server.close()
+        await server.wait_closed()
+        try:
+            await serve_forever
+        except asyncio.CancelledError:
+            # We expect server.close() to emit CancelledError
```

Comment:
this block seems to imply serve_forever emits CancelledError, not server.close

Reply:
`serve_forever` is an `asyncio.Task`, not a method, so awaiting the task effectively joins the thread, raising any exceptions hidden within.

Since we really wanted to communicate that `serve_forever` wasn't an orphaned fork, and that should be handled already by `await wait_closed()`, I replaced it with an `assert ...done()`